packetery/public/admin-country-carrier.js

```diff
@@ -0,0 +1,121 @@
+(function ($) {
+
+    $(function () {
+        // are we on the right page?
+        if ($('.packetery-carrier-options-page').length === 0) {
+            return;
+        }
+
+        var Multiplier = function () {
+            this.registerListeners = function (wrapperSelector) {
+                var $wrappers = $(wrapperSelector);
+                $wrappers
+                    .on('click', '.js-add', function () {
+                        multiplier.addOption(this, $wrappers); // todo use wrapperSelector instead
+                    })
+                    .on('click', '.js-delete', function () {
+                        multiplier.deleteOption(this);
+                    })
+                    .each(function () {
+                        var $wrapper = $(this);
+                        var $inputs = $wrapper.find('input[type="number"]');
+                        // add only if there is something in last row
+                        if ($inputs.eq(-2).val() || $inputs.eq(-1).val()) {
+                            multiplier.addOption($wrapper.find('.js-add'), $wrapper, false);
+                        }
+                        multiplier.toggleDeleteButton($wrapper);
+                    });
+            };
+
+            this.addOption = function (button, $wrappers, setFocus) {
+                var wrapperClassName = $wrappers.first().attr('class'),
+                    $wrapper = $(button).closest('.' + wrapperClassName),
+                    $template = getTemplate($wrapper);
+
+                updateIds($template, newId++);
+                $wrapper.find('table').append($template);
+                if (typeof setFocus === 'undefined') {
+                    setFocus = true;
+                }
+                if (setFocus) {
+                    $('input', $template).eq(0).focus();
+                }
+                this.toggleDeleteButton($wrapper);
+            };
+
+            this.deleteOption = function (button) {
+                var $row = $(button).closest('tr'),
+                    $table = $row.closest('table');
```

Comment:
Nechceme být méně striktní v té HTML struktuře kvůli možnému budoucímu přestylování?

Reply:
A jak si to představuješ? Použít třídu? Je jasné, že JS závislý na DOMu se s úpravou DOMu musí také upravit.